operator/api.go

```diff
@@ -37,16 +35,46 @@ func startServer(addr string, shutdownSignal <-chan struct{}, allSystemsGo <-cha
 		}
 	})
 
-	srv := &http.Server{Addr: addr}
+	errs := make(chan error, 1)
+	nServers := 0
 
-	go func() {
-		<-shutdownSignal
-		if err := srv.Shutdown(context.Background()); err != nil {
-			log.WithError(err).Error("apiserver shutdown")
+	// Since we are opening this on localhost only, we need to make sure
+	// we can open for both v4 and v6 localhost. In case the user is running
+	// v4-only or v6-only.
+	for _, addr := range addrs {
+		if addr == "" {
+			continue
 		}
-	}()
+		nServers++
+		srv := &http.Server{Addr: addr}
+		errCh := make(chan error, 1)
+
+		go func() {
+			err := srv.ListenAndServe()
+			if err != nil {
+				errCh <- err
+				errs <- err
+			}
+		}()
+		go func() {
+			select {
+			case <-shutdownSignal:
+				if err := srv.Shutdown(context.Background()); err != nil {
+					log.WithError(err).Error("apiserver shutdown")
+				}
+			case err := <-errCh:
+				log.Warnf("Unable to start status api: %s", err)
+			}
+		}()
+		log.Infof("Starting apiserver on address %s", addr)
+	}
 
-	log.Fatalf("Unable to start status api: %s", srv.ListenAndServe())
+	for err := range errs {
```

Comment:
Doesn't this race with `errs <- err`? If ListenAndServe fails slow enough, we would have exited this loop. Since it's fatal anyway, we could just put it instead of `errs <- err` (unless you wanted to print an error for each `ListenAndServer` failure, but this loop also races with line 65 I think).

Reply:
nevermind. startServer is not expected to return.